src/sources/file.rs

```diff
@@ -551,7 +551,7 @@ pub fn file_source(
         // The shutdown sent in to the finalizer is the global
         // shutdown handle used to tell it to stop accepting new batch
         // statuses and just wait for the remaining acks to come in.
-        let (finalizer, mut ack_stream) = OrderedFinalizer::<FinalizerEntry>::new(shutdown.clone());
+        let (finalizer, mut ack_stream) = OrderedFinalizer::<FinalizerEntry>::new(None);
```

Comment:
As this is listed as a fix for the file source, it would be good to have a test that demonstrates the situation to demonstrate. Could you work that up?

Reply:
I was able to come up with a test case where a few thousand lines are written to a file, then the file source runs to process it and is shut down, then restarted. 

I'm a little worried it could be a bit of a flaky test, since this is essentially a race between the acknowledgement handling and the shutdown handler, and it will depend on performance of the underlying hardware to some extent. It does fail reliably on my machine when `Some(shutdown.clone())` is passed to the finalizer stream, and passes with `None`.

Reply:
This test run demonstrates the failure of this test: https://github.com/vectordotdev/vector/actions/runs/4555009344/jobs/8033659712?pr=16928